### PR TITLE
Transfer conceptual docs in storybook

### PR DESCRIPTION
**Pull request by mgamis-msft**
# What
Transfer conceptual docs in storybook

# Why
https://skype.visualstudio.com/SPOOL/_workitems/edit/3283389

# How Tested
Ran storybook locally and checked Transfer page loaded with intended content

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/storybook/stories/Transfer.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+requests. Right now only Teams users can request to transfer other participants.
+
+A call with a Teams user can be made by using the `CallParticipantsLocator` as the locator
+in the [AzureCommunicationCallAdapterArgs](./?path=/docs/composite-adapters--page#azurecommunicationcalladapterargs)
+when using the [useAzureCommunicationCallAdapter](./?path=/docs/composite-adapters--page#useazurecommunicationcalladapter)
+function to create your [CallAdapter](path=/docs/composite-adapters--page#call-adapter).
+In the current beta version, only one Teams user can be called.
+
+The id of a Teams user can be obtained by using the
+[Microsoft Graph Explorer](https://developer.microsoft.com/en-us/graph/graph-explorer).
```
Comment (dmceachernmsft) [inline packages/storybook/stories/Transfer.stories.mdx:31]: Is there not documentation that we can link out to here?

File: packages/storybook/stories/Transfer.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+The [CallAdapter](path=/docs/composite-adapters--page#call-adapter) is able to receive `transferRequest` events and
+allows you to handle these events by defining your own `TransferRequestedListener` callback. The following code snippet
+shows an example of a `TransferRequestedListener` that accepts all events.
+
+<Source code={TransferSnippetText} />
+
+The `afterCallAdapterCreate` callback in the above snippet can be modified to handle the transfer request such that audio
+and video states from the original call are maintained after transfer.
+
+```
```
Comment (dmceachernmsft) [inline packages/storybook/stories/Transfer.stories.mdx:56]: nit: lets add some syntax highlighting
```suggestion
```typescript
```

File: packages/storybook/stories/snippets/transfer/Transfer.snippet.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        ? {
+            participantIds: [props.microsoftTeamsUserId]
+          }
+        : undefined
+    }),
+    [props.userId, credential, props.microsoftTeamsUserId]
+  );
+
+  const afterCallAdapterCreate = useCallback(async (adapter: CallAdapter): Promise<CallAdapter> => {
+    adapter.on('transferRequested', (e) => {
```
Comment (dmceachernmsft) [inline packages/storybook/stories/snippets/transfer/Transfer.snippet.tsx:46]: can we change the name from `e` for the event param to `transfer` or something indicative of the object in the event we are actioning `accept` on?

File: packages/storybook/stories/snippets/transfer/TransferWithDialogPrompt.snippet.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        ? {
+            participantIds: [props.microsoftTeamsUserId]
+          }
+        : undefined
+    }),
+    [props.userId, credential, props.microsoftTeamsUserId]
+  );
+
+  const afterCallAdapterCreate = useCallback(async (adapter: CallAdapter): Promise<CallAdapter> => {
+    adapter.on('transferRequested', (e) => {
```
Comment (dmceachernmsft) [inline packages/storybook/stories/snippets/transfer/TransferWithDialogPrompt.snippet.tsx:50]: Same as above

File: packages/storybook/stories/Transfer.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  title="Concepts/Transfer"
+  parameters={{ previewTabs: { canvas: { disable: true, hidden: true } } }}
+/>
+
+# Transfer
+
+<DetailedBetaBanner version={'1.5.2-beta.1'} />
+
+During an active call, participants can transfer another participant to another user or number. The transfer process
+starts when one participant requests a transfer to the another participant which they can accept or reject. In the
```
Comment (edwardlee-msft) [inline packages/storybook/stories/Transfer.stories.mdx:20]:
```suggestion
starts when one participant requests a transfer to another participant where they can accept or reject the call. In the
```

File: packages/storybook/stories/Transfer.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+current beta version, ACS users cannot request to transfer other participants yet but they can receive transfer
+requests. Right now only Teams users can request to transfer other participants.
+
+A call with a Teams user can be made by using the `CallParticipantsLocator` as the locator
+in the [AzureCommunicationCallAdapterArgs](./?path=/docs/composite-adapters--page#azurecommunicationcalladapterargs)
+when using the [useAzureCommunicationCallAdapter](./?path=/docs/composite-adapters--page#useazurecommunicationcalladapter)
+function to create your [CallAdapter](path=/docs/composite-adapters--page#call-adapter).
+In the current beta version, only one Teams user can be called.
+
+The id of a Teams user can be obtained by using the
```
Comment (edwardlee-msft) [inline packages/storybook/stories/Transfer.stories.mdx:30]:
```suggestion
The id of a Teams user can be obtained by using
```

File: packages/storybook/stories/Transfer.stories.mdx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+From there, sign in by clicking the circled button as shown below with a Microsoft account in the same tenant as the
+Teams user you want to call.
+
+<img style={{ width: 'auto', height: 'auto' }} src="images/graph-api-sign-in.png" />
+
+Then create a GET request using v1.0 of the users API like shown below:
+
+<img style={{ width: 'auto', height: 'auto' }} src="images/graph-api-get-user-id-api-request.png" />
+
+The id should be in the API response like highlighted in the example shown below:
```
Comment (edwardlee-msft) [inline packages/storybook/stories/Transfer.stories.mdx:41]:
```suggestion
The id should be in the API response as seen in the example below:
```

File: packages/storybook/stories/Transfer.stories.mdx (hunk 1 above)
Comment (dmceachernmsft) [reply packages/storybook/stories/Transfer.stories.mdx:36]: I also thought you were going more in depth here so happy to resolve this. linking out to this I think is correct

**Review by edwardlee-msft — APPROVED** [review]

**Pull request merged**